source/js/apiHelpers.js

```diff
@@ -155,11 +180,31 @@ async function getRecipesByCuisine(cuisine, num, sort) {
       .then((response) => response.json())
       .then((data) => {
         const ids = extractIDs(data);
-        resolve(getDetailedRecipeInfoBulk(ids));
+        if (!ids) {
+          console.log('No search results');
+          resolve([]);
+        } else {
+          resolve(getDetailedRecipeInfoBulk(ids));
+        }
       })
       .catch((err) => {
-        console.log('Error in searching for recipes by cuisine.');
+        console.log('Error in searching for recipes by type.');
         reject(err);
       });
   });
 }
+
+// export functions?
```

Comment:
yes pls export all of the functions. i think you can look at an example in some of the js files fred or misha wrote to learn the syntax.